apps/web/src/pages/templates/components/email-editor/EmailContentCard.tsx

```diff
@@ -34,8 +40,12 @@ export function EmailContentCard({
     if (integrations.length === 0) {
       return;
     }
-    setIntegration(integrations.find((item) => item.channel === 'email') || null);
-  }, [integrations, setIntegration]);
+    setIntegration(
+      integrations.find((item) =>
+        isMultiProviderConfigEnabled ? item.channel === 'email' && item.selected : item.channel === 'email'
+      ) || null
+    );
```

Comment:
when multi provider feature flag is on then we will find the integration for `email` channel which has the `selected` to `true`